apps/frontend/src/app/PageTeam/CharacterDisplay/Tabs/TabOptimize/Components/BuildDisplayItem.tsx

```diff
@@ -245,23 +245,43 @@ export default function BuildDisplayItem({
   const isActiveBuildOrEquip =
     isActiveBuild || (buildType === 'equipped' && currentlyEquipped)
 
+  // An undefined buildType indicates this was accessed from outside a team. This currently occurs
+  // when comparing from the character editor.
+  const compareFromCharEditor = buildType === undefined
+
   const activeWeapon = useMemo(() => {
     if (dbDirty && buildType === 'real')
       return database.builds.get(buildId)!.weaponId
     if (dbDirty && buildType === 'equipped') return equippedWeapon
+    if (compareFromCharEditor) return equippedWeapon
 
     // default
     return ''
-  }, [dbDirty, database, buildType, buildId, equippedWeapon])
+  }, [
+    dbDirty,
+    database,
+    buildType,
+    buildId,
+    equippedWeapon,
+    compareFromCharEditor,
+  ])
 
   const activeArtifacts = useMemo(() => {
     if (dbDirty && buildType === 'real')
       return database.builds.get(buildId)!.artifactIds
     if (dbDirty && buildType === 'equipped') return equippedArtifacts
+    if (compareFromCharEditor) return equippedArtifacts
```

Comment:
I'd recommend hoisting this up to the first line in the function

Reply:
done